src/jit/jitconfigvalues.h

```diff
@@ -0,0 +1,193 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
> Just widening policy used by getXConfigValue() JIT-EE call won't be sufficient for the following reasons:
>
> consider the following config value from clrconfigvalues.h
> 
> RETAIL_CONFIG_DWORD_INFO_EX(INTERNAL_UseRyuJit, W("UseRyuJIT"), 0, "Set to 1 by .NET 4.6 installer to indicate RyuJIT should be used, not JIT64.", CLRConfig::IgnoreEnv | CLRConfig::IgnoreHKCU | CLRConfig::IgnoreConfigFiles)
> 
> Look at the config options: this specific setting cannot come from env, cannot be under HKCU nor from config files.

The keys you've mentioned so far are not read by RyuJIT They are both read by the EE and therefore do not apply to this discussion. I've put a list of the config values read by RyuJIT, their current policies, and whether or not they are present in retail builds here: https://gist.github.com/pgavlin/fc177a79c89abb51fe28.

> We cannot suddenly allow some config settings to come from sources which are not supported previously for a reason (e.g. requiring admin privileges for modifying certain reg hive keys etc).

The only widening that would happen here would be allowing options whose values could not previously come from a config file to now come from said file. This could be a problem if there exist config files that contain values for options that wouldn't have read them before. This is certainly a risk, but it seems rather rare to me.

> Further in future while adding more config values, we would like to have such a flexibility.

Part of the motivation behind accessing configuration values using the JIT interface is to move policy decisions (e.g. which config values are accessible and from where) out of the JIT. While I'm open to modifying the API s.t. it is easier for the EE to make policy decisions (e.g. by using enum values rather than strings as previously mentioned), I do think that we should try to keep the APIs themselves policy-free.